sdk/eventhubs/azure-messaging-eventhubs/README.md

```diff
@@ -1,3 +1,186 @@
+The Processor is useful when you want to have the partition assignment be dynamically chosen, and balanced with other Processor instances.
+
+More information about Event Hubs features and terminology can be found here: [link](https://docs.microsoft.com/azure/event-hubs/event-hubs-features)
+
+
+# Examples
+
+<!-- NOTE: Fix dead links -->
+Examples for various scenarios can be found on [azure.github.io](https://azure.github.io/azure-sdk-for-cpp/storage.html) or in the samples directory in our GitHub repo for 
+[EventHubs](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/eventhubs).
+
+## Send events
+
+The following example shows how to send events to an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+Azure::Messaging::EventHubs::EventDataBatchOptions batchOptions;
+batchOptions.PartitionId = "1";
+Azure::Messaging::EventHubs::EventDataBatch eventBatch(batchOptions);
+
+Azure::Messaging::EventHubs::Models::EventData message;
+message.Body.Data = {'H', 'e', 'l', 'l', 'o', '2'};
+
+eventBatch.AddMessage(message);
+
+Azure::Messaging::EventHubs::ProducerClientOptions producerOptions;
+producerOptions.Name = "sender-link";
+producerOptions.ApplicationID = "some";
+
+auto client = Azure::Messaging::EventHubs::ProducerClient(
+      connectionString, eventHubName, producerOptions);
+auto result = client.SendEventDataBatch(eventBatch);
+```
+
+## Receive events
+
+The following example shows how to receive events from partition 1 on an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+auto client = Azure::Messaging::EventHubs::ConsumerClient(
+	connectionString, eventHubName);
+
+Azure::Messaging::EventHubs::PartitionClientOptions partitionOptions;
+// We want to consume all messages from the earliest (the default behavior is to consume new events).
+partitionOptions.StartPosition.Inclusive = true;
+partitionOptions.StartPosition.Earliest = true;
+Azure::Messaging::EventHubs::PartitionClient partitionClient
+        = client.CreatePartitionClient("1", partitionOptions);
+
+auto events = partitionClient.ReceiveEvents(1);
+```
+
+
+# Troubleshooting
+
+## Logging
+
+1. The EventHubs SDK client uses the [Azure SDK log message](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/core/azure-core#sdk-log-messages) functionality to 
+enable diagnostics.
+1. In addition to the direct traces, the [ProducerClient][producer_client] and [ConsumerClient][consumer_client] classes also support logging of the events that are being sent/received.
+
+To enable additional logging, you can enable the `VerboseLogging` client option field:
```

Comment:
We discussed offline to remove this option field for now.

In case we decide to bring this option back, I recommend a name like `AllowLogging` to match the semantics, since it on its own doesn't actually enable logging, but rather allows for any logging enabled via setting of the log level, to be emitted. And using verbose in the option name overloads the log_level we have, which leads to confusion.